main.go

```diff
@@ -2,41 +2,58 @@ package main
 
 import (
 	"database/sql"
+	"flag"
 	"fmt"
 	"net/http"
 	"os"
+	"strconv"
+	"strings"
 	"time"
 
 	_ "github.com/mattn/go-sqlite3"
 	"github.com/op/go-logging"
 )
 
+const (
+	portDefault = "8080"
+)
+
 var logger = logging.MustGetLogger("commento")
 var db *sql.DB
 
+// The run environment for the application. Either dev, staging, or prod
+var runEnv string
+
 func main() {
-	err := loadDatabase("sqlite3.db")
-	if err != nil {
+	if err := loadDatabase("sqlite3.db"); err != nil {
 		die(err)
 	}
 
-	fs := http.FileServer(http.Dir("assets"))
+	// Parse command line options
+	var port = flag.String("port", portDefault, "port for commento service")
+	flag.Parse()
 
-	http.Handle("/assets/", http.StripPrefix("/assets/", fs))
-	http.HandleFunc("/", indexHandler)
-	http.HandleFunc("/create", createCommentHandler)
-	http.HandleFunc("/get", getCommentsHandler)
-
-	var port string
+        // Check that we get a valid port value
+	if _, err := strconv.ParseInt(*port, 10, 32); err != nil {
+		die(err)
+	}
 
-	if fromEnv := os.Getenv("PORT"); fromEnv != "" {
-		port = ":" + fromEnv
-	} else {
-		port = ":8080"
+	// Check for the run environment, default to dev
+	runEnv = strings.ToLower(os.Getenv("RUN_ENV"))
```

Comment:
Most of the run state will be passed in environment variables in accordance with the 12 factor philosophy. Passing PORT as an environment variable might cause problems  according to @DarkThrone so it will be one of the few exceptions.